cloudbridge/cloud/providers/aws/resources.py

```diff
@@ -858,19 +856,24 @@ def get(self, key):
         Retrieve a given object from this bucket.
         """
         key = Key(self._bucket, key)
-        if key.exists():
+        if key and key.exists():
             return AWSBucketObject(self._provider, key)
         return None
 
-    def list(self, limit=None, marker=None):
+    def list(self, limit=None, marker=None, prefix=None):
         """
         List all objects within this bucket.
 
         :rtype: BucketObject
         :return: List of all available BucketObjects within this bucket.
         """
-        objects = [AWSBucketObject(self._provider, obj)
-                   for obj in self._bucket.list()]
+        if prefix:
```

Comment:
I think you can omit this `if` here and just pass `prefix` regardless because if it's `None` it will just default to boto's default value anyhow.